// pcln-design-system/dist/es/BreadcrumbLink.js
import React from 'react';
import PropTypes from 'prop-types';
import Flex from './Flex';
import Link from './Link';
import Text from './Text';

function BreadcrumbLink(_ref) {
  var className = _ref.className,
      isLastChild = _ref.isLastChild,
      href = _ref.href,
      icon = _ref.icon,
      label = _ref.label,
      onClick = _ref.onClick;
  var linkColor = isLastChild ? 'text.dark' : 'text.light';
  return React.createElement(Flex, {
    className: className,
    alignItems: "center"
  }, icon, React.createElement(Link, {
    href: href,
    color: linkColor,
    onClick: onClick
  }, label), !isLastChild && React.createElement(Text, {
    m: 2,
    color: "text.light"
  }, "/"));
}

BreadcrumbLink.displayName = 'BreadcrumbLink';
BreadcrumbLink.propTypes = {
  className: PropTypes.string,
  isLastChild: PropTypes.bool,
  href: PropTypes.string,
  icon: PropTypes.node,
  label: PropTypes.string,
  onClick: PropTypes.func
};
BreadcrumbLink.defaultProps = {
  className: ''
};
export default BreadcrumbLink;
