// pcln-design-system/dist/cjs/Checkbox.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _pclnIcons = require("pcln-icons");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Box = _interopRequireDefault(require("./Box"));

var _utils = require("./utils");

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  appearance: none;\n  opacity: 0;\n  position: absolute;\n  z-index: 0;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-flex;\n  align-items: center;\n  position: relative;\n  vertical-align: middle;\n  padding: 2px;\n  cursor: pointer;\n  background-color: inherit;\n  color: ", ";\n\n  svg {\n    border: 1px solid transparent;\n    border-radius: 4px;\n    padding: 2px;\n  }\n\n  svg[data-name='checked'] {\n    display: none;\n  }\n  \n  > input:hover ~ svg[data-name='empty'] {\n    color: ", ";\n      }\n  }\n  \n  > input {\n    &:focus ~ svg {\n      border: 1px solid ", ";\n      background-color: ", ";\n    }\n  }\n\n  > input:checked {\n    & ~ svg[data-name='checked'] {\n      display: inline-block;\n      color: ", ";\n    }\n\n    & ~ svg[data-name='empty'] {\n      display: none;\n    }\n    \n    &:focus ~ svg {\n      border: 1px solid ", ";\n      background-color: ", ";\n    }\n\n    &:hover ~ svg[data-name='checked'] {\n      color: ", ";\n        }\n  }\n\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Checkbox = function Checkbox(props) {
  var disabled = props.disabled,
      size = props.size; // Add 4px to Icon's height and width to account for size reduction caused by adding padding to SVG element

  var borderAdjustedSize = size + 4;
  return _react["default"].createElement(CheckBoxWrapper, {
    theme: props.theme,
    color: props.color,
    disabled: disabled
  }, _react["default"].createElement(StyledInput, (0, _extends2["default"])({
    type: "checkbox"
  }, props, {
    role: "checkbox"
  })), _react["default"].createElement(_pclnIcons.BoxChecked, {
    size: borderAdjustedSize,
    "data-name": "checked"
  }), _react["default"].createElement(_pclnIcons.BoxEmpty, {
    size: borderAdjustedSize,
    "data-name": "empty"
  }));
};

var CheckBoxWrapper = (0, _styledComponents["default"])(_Box["default"])(_templateObject(), function (props) {
  return props.disabled ? (0, _utils.getPaletteColor)('border.base')(props) : (0, _utils.getPaletteColor)('border.light')(props);
}, function (props) {
  return props.disabled ? (0, _utils.getPaletteColor)('border.base')(props) : (0, _utils.getPaletteColor)('base')(props);
}, (0, _utils.getPaletteColor)('border.base'), (0, _utils.getPaletteColor)('background.light'), function (props) {
  return props.disabled ? (0, _utils.getPaletteColor)('border.base')(props) : (0, _utils.getPaletteColor)('base')(props);
}, (0, _utils.getPaletteColor)('base'), (0, _utils.getPaletteColor)('light'), function (props) {
  return props.disabled ? props.theme.colors.borderGray : props.theme.colors.darkBlue;
}, (0, _utils.applyVariations)('Checkbox'));

var StyledInput = _styledComponents["default"].input(_templateObject2());

Checkbox.displayName = 'Checkbox';
Checkbox.propTypes = {
  id: _propTypes["default"].string.isRequired,
  size: _propTypes["default"].number,
  onChange: _propTypes["default"].func.isRequired,
  color: (0, _utils.deprecatedColorValue)()
};
Checkbox.defaultProps = {
  size: 20,
  color: 'primary'
};

var _default = (0, _styledComponents.withTheme)(Checkbox);

exports["default"] = _default;
