// pcln-design-system/dist/cjs/Stamp.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _utils = require("./utils");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-flex;\n  align-items: center;\n  vertical-align: top;\n  min-height: 24px;\n  font-weight: 600;\n  text-transform: uppercase;\n  letter-spacing: ", ";\n  border-radius: 2px;\n  border-width: 1px;\n  border-style: solid;\n  border-color: ", ";\n  color: ", ";\n  ", "\n  \n  ", " ", ";\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Stamp = _styledComponents["default"].div(_templateObject(), (0, _styledSystem.themeGet)('letterSpacings.caps'), function (props) {
  return (0, _utils.getPaletteColor)(props.borderColor, 'base')(props);
}, (0, _utils.getPaletteColor)('base'), function (props) {
  return props.bg ? "background-color: ".concat((0, _utils.getPaletteColor)(props.bg, 'base')(props), ";") : '';
}, _styledSystem.space, _styledSystem.fontSize, (0, _utils.applyVariations)('Stamp'));

Stamp.displayName = 'Stamp';
Stamp.propTypes = _objectSpread({}, _styledSystem.space.propTypes, {}, _styledSystem.fontSize.propTypes, {
  color: (0, _utils.deprecatedColorValue)(),
  bg: (0, _utils.deprecatedPropType)('color'),
  borderColor: (0, _utils.deprecatedColorValue)()
});
Stamp.defaultProps = {
  px: 1,
  py: 0,
  color: 'border.light',
  bg: 'background.light',
  borderColor: 'border.base',
  fontSize: 0
};
var _default = Stamp;
exports["default"] = _default;
